src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/DefaultRazorSemanticTokenInfoServiceTest.cs

```diff
@@ -6,314 +6,445 @@
+            await AssertSemanticTokensAsync(documentText, isRazorFile: false, razorRange, csharpTokens: csharpTokens);
         }
 
         [Fact]
         public async Task GetSemanticTokens_MinimizedHTMLAsync()
         {
-            var txt = $"@addTagHelper *, TestAssembly{Environment.NewLine}<input/> ";
+            var documentText = @"@addTagHelper *, TestAssembly
+<input/>
+";
+
+            var razorRange = new OS.Range
+            {
+                Start = new OS.Position { Line = 0, Character = 0 },
+                End = new OS.Position { Line = 2, Character = 0 }
+            };
 
-            await AssertSemanticTokensAsync(txt, isRazor: false);
+            var csharpTokens = await GetCSharpSemanticTokensResponseAsync(documentText, razorRange, isRazorFile: false);
+            await AssertSemanticTokensAsync(documentText, isRazorFile: false, razorRange, csharpTokens: csharpTokens);
         }
 
         [Fact]
         public async Task GetSemanticTokens_HTMLCommentAsync()
         {
-            var txt = $"@addTagHelper *, TestAssembly{Environment.NewLine}<!-- comment with comma's --> ";
+            var documentText = @"@addTagHelper *, TestAssembly
+<!-- comment with comma's -->
+";
 
-            await AssertSemanticTokensAsync(txt, isRazor: false);
+            var razorRange = new OS.Range
+            {
+                Start = new OS.Position { Line = 0, Character = 0 },
+                End = new OS.Position { Line = 2, Character = 0 }
+            };
+
+            var csharpTokens = await GetCSharpSemanticTokensResponseAsync(documentText, razorRange, isRazorFile: false);
+            await AssertSemanticTokensAsync(documentText, isRazorFile: false, razorRange, csharpTokens: csharpTokens);
         }
 
         [Fact]
         public async Task GetSemanticTokens_PartialHTMLCommentAsync()
         {
-            var txt = $"@addTagHelper *, TestAssembly{Environment.NewLine}<!-- comment";
+            var documentText = @"@addTagHelper *, TestAssembly
+<!-- comment
+";
 
-            await AssertSemanticTokensAsync(txt, isRazor: true);
+            var razorRange = new OS.Range
+            {
+                Start = new OS.Position { Line = 0, Character = 0 },
+                End = new OS.Position { Line = 2, Character = 0 }
+            };
+
+            var csharpTokens = await GetCSharpSemanticTokensResponseAsync(documentText, razorRange, isRazorFile: true);
+            await AssertSemanticTokensAsync(documentText, isRazorFile: true, razorRange, csharpTokens: csharpTokens);
         }
 
         [Fact]
         public async Task GetSemanticTokens_HTMLIncludesBang()
         {
-            var txt = $"@addTagHelper *, TestAssembly{Environment.NewLine}<!input/>";
+            var documentText = @"@addTagHelper *, TestAssembly
+<!input/>
+";
 
-            await AssertSemanticTokensAsync(txt, isRazor: false);
+            var razorRange = new OS.Range
```

Comment:
I'm not a big fan of how explicit always having to pass the range is, but I suppose we're specifically testing the RANGE handler so it make sense.